From 9ed722e4c1dd1bef5742b9413fc42cdc0398e7fc Mon Sep 17 00:00:00 2001
From: Francois Killeen <killeenfrancois@gmail.com>
Date: Sat, 23 Mar 2019 16:20:30 -0400
Subject: [PATCH] libnl3fix

---
 wpa_supplicant/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/wpa_supplicant/Makefile b/wpa_supplicant/Makefile
index 6d0e381..985cfcc 100644
--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -134,6 +134,11 @@ NEED_SHA256=y
 NEED_AES_OMAC1=y
 endif
 
+ifdef CONFIG_LIBNL20
+LIBS += -lnl-3
+CFLAGS += -DCONFIG_LIBNL20
+endif
+
 ifdef CONFIG_IEEE80211R
 CFLAGS += -DCONFIG_IEEE80211R
 OBJS += ../src/rsn_supp/wpa_ft.o

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 364158c..f3fafbd 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -65,6 +65,14 @@ struct i802_bss {
        unsigned int beacon_set:1;
 };
 
+#ifdef CONFIG_LIBNL20
+/* libnl 2.0 compatibility code */
+
+#define nl_handle nl_sock
+#define nl_handle_alloc_cb nl_socket_alloc_cb
+#define nl_handle_destroy nl_socket_free
+#endif /* CONFIG_LIBNL20 */
+
 struct wpa_driver_nl80211_data {
        void *ctx;
        struct netlink_data *netlink;
diff --git a/src/drivers/drivers.mak b/src/drivers/drivers.mak
index b76b229..b37e243 100644
--- a/src/drivers/drivers.mak
+++ b/src/drivers/drivers.mak
@@ -31,10 +31,10 @@ NEED_SME=y
 NEED_AP_MLME=y
 NEED_NETLINK=y
 NEED_LINUX_IOCTL=y
-DRV_LIBS += -lnl
+DRV_LIBS += -lnl-3
 
 ifdef CONFIG_LIBNL20
-DRV_LIBS += -lnl-genl
+DRV_LIBS += -lnl-genl-3
 DRV_CFLAGS += -DCONFIG_LIBNL20
 endif
 endif
